src/h5web/providers/hsds/api.ts

```diff
@@ -57,51 +58,18 @@ export class HsdsApi implements ProviderAPI {
   public async getMetadata(): Promise<Metadata> {
     const rootId = await this.getRootId();
     await this.processGroup(rootId);
-
-    return buildTree(
-      {
-        root: rootId,
-        groups: this.groups,
-        datasets: this.datasets,
-        datatypes: this.datatypes,
-      },
-      this.domain
-    );
+    return buildTree(await this.getHdf5Metadata(), this.domain);
   }
 
   public async getGroup(path: string): Promise<Group> {
     if (this.groupsByPath.has(path)) {
       return this.groupsByPath.get(path) as Group;
     }
 
-    if (path === '/') {
-      const rootId = await this.getRootId();
-      const rootGroup: Group = {
-        uid: nanoid(),
-        name: this.domain,
-        id: rootId,
-        kind: EntityKind.Group,
-        attributes: [],
-        children: await this.getGroupChildren(rootId),
-      };
-
-      this.groupsByPath.set('/', rootGroup);
-      return rootGroup;
-    }
-
-    const parentPath = path.slice(0, path.lastIndexOf('/')) || '/';
-    const parentGroup = await this.getGroup(parentPath);
-
-    const childName = path.slice(path.lastIndexOf('/') + 1);
-    const child = parentGroup.children.find(({ name }) => name === childName);
-    assertDefined(child);
-    assertGroup(child);
-
-    const group: Group = {
-      ...child,
-      children: await this.getGroupChildren(child.id),
-    };
+    const groupLink = await this.getLinkTo(path);
+    await this.processGroup(groupLink.id, 1);
 
+    const group = buildGroup(await this.getHdf5Metadata(), groupLink);
```

Comment:
Quite clever to delegate the handling of non-resolvable links to `buildGroup`.

Unfortunately, it means that the transformation from a `HSDSLink` to a `Link` is no longer made (previously made in `getGroupChildren` :
```
  const link = isHsdsExternalLink(hsdsLink)
    ? { ...hsdsLink, file: hsdsLink.h5domain }
    : hsdsLink;
```
)

Reply:
This is now done by `processGroup`, like it always should have been. I wasn't calling it in my early prototype.